receiver/awscontainerinsightreceiver/internal/host/ec2metadata.go

```diff
@@ -0,0 +1,96 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package host
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsec2metadata "github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+
+	"go.uber.org/zap"
+)
+
+type Ec2metadata struct {
+	logger          *zap.Logger
+	refreshInterval time.Duration
+	instanceID      string
+	instanceType    string
+	shutdownC       chan bool
+}
+
+func NewEc2metadata(refreshInterval time.Duration, logger *zap.Logger) *Ec2metadata {
+	emd := &Ec2metadata{
+		refreshInterval: refreshInterval,
+		shutdownC:       make(chan bool),
+		logger:          logger,
+	}
+
+	emd.refresh()
+	go func() {
+		refreshTicker := time.NewTicker(emd.refreshInterval)
+		defer refreshTicker.Stop()
+		for {
+			select {
+			case <-refreshTicker.C:
+				//stop the refresh once we get instance ID and type successfully
```

Comment:
Can you use aws sdk's `retryer` instead of implementing this yourself? Same for all similar loops

Reply:
`retryer` seems to be a global retry policy on aws config https://github.com/aws/aws-sdk-go/blob/main/aws/request/retryer.go though I guess it's possible to inspect the request and choose different retry time for different request.

Reply:
The retryer interface requires to implement `MaxRetries()`: https://github.com/aws/aws-sdk-go/blob/main/aws/request/retryer.go#L32-L34. This is not aligned with the use case here because we always want the retry to happen until its succeeds (no matter the number of retries). I don't see a way to set the max retries to infinite.

Reply:
I have refactored out the retry logic into an util function